File: app/src/main/java/com/itmoldova/sync/SyncScheduler.java
package com.itmoldova.sync;

/**
 * Defines the interface for scheduling repeating syncs.
 */
public interface SyncScheduler {

    /**
     * Schedule repeating sync at the specified interval.
     */
    void scheduleRepeatingSync(long interval);

    /**
     * Cancel sync
     */
    void cancel();

}

File: app/src/main/java/com/itmoldova/sync/SyncScheduler.kt
package com.itmoldova.sync

/**
 * Defines the interface for scheduling repeating syncs.
 */
interface SyncScheduler {

    /**
     * Schedule repeating sync at the specified interval.
     */
    fun scheduleRepeatingSync(interval: Long)

    /**
     * Cancel sync
     */
    fun cancel()

}
